gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/traverser/B_O_Traverser.java

```diff
@@ -62,10 +62,12 @@ public void setStepId(final String stepId) {
         this.future = stepId;
     }
 
+    protected final boolean equals(final B_O_Traverser other) {
+        return super.equals(other) && other.future.equals(this.future);
+    }
+
     @Override
```

Comment:
This class does not have a `hashCode()` method. It should since it implements `equals()`. Its easy, its just `this.t.hashCode()`.

Reply:
That's intentional. B_O_Traverser's hashcode is the same as O_Traverser's. Same story below. In both cases equals() will do the "deep" check, but hashcode() ignores fields that are unlikely different.

Reply:
I agree with @okram. There should be an explicit `hashCode()` so it doesn't look like it is mistakenly forgotten.